src/ntnx/cartesian/Brush.js

```diff
@@ -0,0 +1,188 @@
+    /**
+     * The height of brush.
+     */
+    height: PropTypes.number.isRequired,
+    /**
+     * The width of each traveller.
+     */
+    travellerWidth: PropTypes.number,
+    /**
+     * The data with gap of refreshing chart. If the option is not set, the chart will be refreshed every time.
+     */
+    gap: PropTypes.number,
+    /**
+     * Padding.
+     */
+    padding: PropTypes.shape({
+      top: PropTypes.number,
+      right: PropTypes.number,
+      bottom: PropTypes.number,
+      left: PropTypes.number,
+    }),
+    /**
+     * The key of data displayed in Brush.
+     */
+    dataKey: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.func]),
+    /**
+     * The original data of a LineChart, a BarChart or an AreaChart.
+     */
+    data: PropTypes.array,
+    /**
+     * The default start index of brush. If the option is not set, the start index will be 0.
+     */
+    startIndex: PropTypes.number,
+    /**
+     * The default end index of brush. If the option is not set, the end index will be calculated by the length of data.
+     */
+    endIndex: PropTypes.number,
+    /**
+     * The formatter function of ticks.
+     */
+    tickFormatter: PropTypes.func,
+    /**
+     * Children.
+     */
+    children: PropTypes.node,
+    /**
+     * The handler of changing the active scope of brush.
+     */
+    onChange: PropTypes.func,
+    /**
+     * 
+     */
+    updateId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    /**
+     * Leave time out.
+     */
+    leaveTimeOut: PropTypes.number,
+  };
+
+  static defaultProps = {
+    height: 15,
+    travellerWidth: 5,
+    gap: 1,
+    fill: '#fff',
+    stroke: 'none',
+    padding: { top: 1, right: 1, bottom: 1, left: 1 },
+    leaveTimeOut: 1000,
+  };
+
+  /**
+   * Render Slider Bar
+   * @param {number} startX - x coordinate to start render slider
+   * @param {number} endX - x coordinate to end render slider
+   * @returns {element}
+   */
+  renderSlide(startX, endX) {
+    const { y, height, stroke } = this.props;
+    const sliderHeight = 4;
+    const sliderPos = y + Math.ceil(height / 2) -  Math.ceil(sliderHeight / 2);
+    return (
+      <Layer className="brush-slider-group">
+        <rect
+          className="recharts-brush-slide ntnx"
+          fill={ ThemeManager.getVar('light-gray-3') }
+          x={Math.min(startX, endX)}
+          y={ sliderPos }
+          rx={ 4 }
+          ry={ 4 }
+          width={Math.abs(endX - startX)}
+          height={ sliderHeight }
+        />
+        <rect
+          className="recharts-brush-slide-grabber ntnx"
+          onMouseEnter={this.handleEnterSlideOrTraveller}
+          onMouseLeave={this.handleLeaveSlideOrTraveller}
+          onMouseDown={this.handleSlideDragStart}
+          onTouchStart={this.handleSlideDragStart}
+          style={{ cursor: 'move' }}
+          fill='rgba(255, 255, 255, 0)'
+          x={Math.min(startX, endX)}
+          y={y}
```

Comment:
Is linting on for this project? No spaces in between brackets here, but there are spaces on the next line. Also noticed some other linting inconsistencies in various places

Reply:
No linting installed. When We move this to the nutanix-core. I will install linting and fix all the inconsistencies.